docker/pravega/scripts/lts_recovery_backup.sh

```diff
@@ -0,0 +1,87 @@
+max_attempt=3
+home_dir=/opt/pravega
+
+# validate method accepts the below three parameters.
+# $1 is the output of grep command.
+# $2 is the value that we are not expecting in the output.
+# $3 either contains 'Pravega cluster' or 'Zookeeper client'.
+validate() {
+  # if $2 == $1 that means output is empty.
+  if [ "$2" == "$1" ]
+  then
+    echo "Failed to get " $3
+    exit 1
+  fi
+}
+
+val=''
+
+# get_val method accepts the below two parameters.
+# $1 Property keys for which the value needs to be searched.
+# $2 either contains 'Pravega cluster' or 'Zookeeper client'.
+get_val() {
+  cmd=$1
+  # Greps the key matching for parameter $1
+  output=$(ps -aef | more | grep -o $cmd)
+  # Validates the output. If nothing is found in grep then it returns the parameter that we are searching.
+  validate "$output" "${cmd:1}" "$2"
+  # Split based on space and get first column.
+  output=$(echo $output | cut -d ' ' -f1)
+  # Validating output whether it is empty or not.
+  validate "$output" "" "$2"
+  # Splitting based on '=' to get the value for the key.
+  val=$(echo $output | cut -d '=' -f2)
+  # Validating value whether it is empty or not.
+  validate "$val" "" "$2"
+}
+
+set_configuration() {
+  get_val "\Dpravegaservice.clusterName=.*D" "Pravega cluster"
```

Comment:
i think this can be much simpler...that is `printenv | grep "CLUSTER_NAME`

Reply:
Fixed it.